gcloud-java-contrib/pom.xml

```diff
@@ -35,6 +35,44 @@
           <skip>false</skip>
         </configuration>
       </plugin>
+        <!-- make a shaded fat jar so we can demo adding the NIO jar to add functionality. -->
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <version>2.4.3</version>
+            <configuration>
+                <!-- put your configurations here -->
+            </configuration>
+            <executions>
+                <execution>
+                    <phase>package</phase>
+                    <goals>
+                        <goal>shade</goal>
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
```

Comment:
why do we need the assembly-plugin? I would rather have only the shaded far jar option.
Also, can you make the shaded jar artifcat's name more obvious that it is a shaded jar all?